pkg/audit/verifier.go

```diff
@@ -487,23 +511,46 @@ func (verifier *Verifier) GetShare(ctx context.Context, limit *pb.AddressedOrder
 	}, nil
 }
 
+// RemoveFailedPieces removes lost pieces from a pointer
+func (verifier *Verifier) RemoveFailedPieces(ctx context.Context, path string, pointer *pb.Pointer, failedNodes storj.NodeIDList) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	if len(failedNodes) == 0 {
+		return nil
+	}
+	remoteSegment := pointer.GetRemote()
+	newRemotePieces := remoteSegment.RemotePieces[:0]
+OUTER:
+	for _, piece := range remoteSegment.RemotePieces {
+		for _, failedNode := range failedNodes {
+			if piece.NodeId == failedNode {
+				continue OUTER
+			}
+		}
+		newRemotePieces = append(newRemotePieces, piece)
+	}
+	remoteSegment.RemotePieces = newRemotePieces
+
+	// Update the segment pointer in the metainfo
+	//TODO:  update in a safe manner - https://storjlabs.atlassian.net/browse/V3-2088
+	return verifier.metainfo.Put(ctx, path, pointer)
```

Comment:
we might need to wrap this error but honestly I'm never super sure when wrapping is required lol

Reply:
So far it should be fine not to wrap it, it's used internally and the lines that call this function don't return it, they just log the error, so we'll get the line from the log and we'll know that it comes from this method.

I've seen other places in this package where wrapping the errors is more important than this one.

Reply:
cool, thanks for the clarification!